packages/asset-swapper/src/utils/aggregate.ts

```diff
@@ -0,0 +1,780 @@
+            orders.push((fill.data as NativeFillData).order);
+        } else if (source === ERC20BridgeSource.Eth2Dai) {
+            orders.push(createEth2DaiOrder(inputToken, outputToken, fill.input, fill.output, bridgeSlippage, true));
+        } else if (source === ERC20BridgeSource.Uniswap) {
+            orders.push(createUniswapOrder(inputToken, outputToken, fill.input, fill.output, bridgeSlippage, true));
+        } else {
+            throw new Error(`invalid buy fill source: ${source}`);
+        }
+    }
+    return orders;
+}
+
+function createKyberOrder(
+    makerToken: string,
+    takerToken: string,
+    makerAssetAmount: BigNumber,
+    takerAssetAmount: BigNumber,
+    slippage: number,
+    isBuy: boolean = false,
+): OrderWithoutDomain {
+    return createBridgeOrder(
+        KYBER_BRIDGE_ADDRESS,
+        makerToken,
+        takerToken,
+        makerAssetAmount,
+        takerAssetAmount,
+        slippage,
+        isBuy,
+    );
+}
+
+function createEth2DaiOrder(
+    makerToken: string,
+    takerToken: string,
+    makerAssetAmount: BigNumber,
+    takerAssetAmount: BigNumber,
+    slippage: number,
+    isBuy: boolean = false,
+): OrderWithoutDomain {
+    return createBridgeOrder(
+        ETH2DAI_BRIDGE_ADDRESS,
+        makerToken,
+        takerToken,
+        makerAssetAmount,
+        takerAssetAmount,
+        slippage,
+        isBuy,
+    );
+}
+
+function createUniswapOrder(
+    makerToken: string,
+    takerToken: string,
+    makerAssetAmount: BigNumber,
+    takerAssetAmount: BigNumber,
+    slippage: number,
+    isBuy: boolean = false,
+): OrderWithoutDomain {
+    return createBridgeOrder(
+        UNISWAP_BRIDGE_ADDRESS,
+        makerToken,
+        takerToken,
+        makerAssetAmount,
+        takerAssetAmount,
+        slippage,
+        isBuy,
+    );
+}
+
+function createBridgeOrder(
+    bridgeAddress: string,
+    makerToken: string,
+    takerToken: string,
+    makerAssetAmount: BigNumber,
+    takerAssetAmount: BigNumber,
+    slippage: number,
+    isBuy: boolean = false,
+): OrderWithoutDomain {
+    return {
+        makerAddress: bridgeAddress,
+        makerAssetData: createBridgeAssetData(makerToken, bridgeAddress, createBridgeData(takerToken)),
+        takerAssetData: createERC20AssetData(takerToken),
+        ...createCommonOrderFields(makerAssetAmount, takerAssetAmount, slippage, isBuy),
+    };
+}
+
+/**
+ * Create ERC20Bridge asset data.
+ */
+export function createBridgeAssetData(tokenAddress: string, bridgeAddress: string, bridgeData: string): string {
+    const encoder = AbiEncoder.createMethod('ERC20Bridge', [
+        { name: 'tokenAddress', type: 'address' },
+        { name: 'bridgeAddress', type: 'address' },
+        { name: 'bridgeData', type: 'bytes' },
+    ]);
+    return encoder.encode({ tokenAddress, bridgeAddress, bridgeData });
+}
+
+/**
+ * Create ERC20Proxy asset data.
+ */
+export function createERC20AssetData(tokenAddress: string): string {
```

Comment:
we can re-use `DevUtils` here which performs this `pure` in memory.

Reply:
I hate `DevUtils` but OK :roll_eyes:

Reply:
On second thought, how about we wait until `DevUtils` is sorted out since switching this one function won't really do me any favors right now.